transport/http/src/test/java/org/kaazing/gateway/transport/http/acceptor/specification/rfc7230/MessageFormatIT.java

```diff
@@ -0,0 +1,423 @@
+            @Override
+            protected void doSessionOpened(HttpAcceptSession session) throws Exception {
+                latch.countDown();
+                session.setStatus(HttpStatus.CLIENT_BAD_REQUEST);
+                session.close(false);
+            }
+        };
+        acceptor.bind(HTTP_ADDRESS, acceptHandler);
+
+        k3po.finish();
+        assertTrue(latch.await(4, SECONDS));
+    }
+
+    @Test
+    @Specification({"server.should.send.501.to.unimplemented.methods/request"})
+    public void serverShouldSend501ToUnImplementedMethods() throws Exception {
+        final IoHandler acceptHandler = new IoHandlerAdapter<HttpAcceptSession>();
+        acceptor.bind(HTTP_ADDRESS, acceptHandler);
+
+        k3po.finish();
+    }
+
+    @Ignore("BUG: No response to error in URI")
+    @Test
+    @Specification({"server.should.send.414.to.request.with.too.long.a.request/request"})
+    public void serverShouldSend414ToRequestWithTooLongARequest() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final IoHandler acceptHandler = new IoHandlerAdapter<HttpAcceptSession>() {
+            @Override
+            protected void doSessionOpened(HttpAcceptSession session) throws Exception {
+                latch.countDown();
+                session.setStatus(HttpStatus.CLIENT_REQUEST_URI_TOO_LONG);
+                session.close(false);
+            }
+        };
+        acceptor.bind(HTTP_ADDRESS, acceptHandler);
+
+        k3po.finish();
+        assertTrue(latch.await(4, SECONDS));
+    }
+
+    @Test
+    @Specification({"server.should.send.status.line.in.start.line/request"})
+    public void serverShouldSendStatusLineInStartLine() throws Exception {
+        standardHttpTestCase(HTTP_ADDRESS);
+    }
+
+    /*
+     * No whitespace is allowed between the header field-name and colon.  In
+     * the past, differences in the handling of such whitespace have led to
+     * security vulnerabilities in request routing and response handling.  A
+     * server MUST reject any received request message that contains
+     * whitespace between a header field-name and colon with a response code
+     * of 400 (Bad Request).  A proxy MUST remove any such whitespace from a
+     * response message before forwarding the message downstream.
+     */
+    @Ignore("Not Spec Complient")
+    @Test
+    @Specification({"server.must.reject.header.with.space.between.header.name.and.colon/request"})
+    public void serverMustRejectHeaderWithSpaceBetweenHeaderNameAndColon() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final IoHandler acceptHandler = new IoHandlerAdapter<HttpAcceptSession>() {
+            @Override
+            protected void doSessionOpened(HttpAcceptSession session) throws Exception {
+                latch.countDown();
+                session.setStatus(HttpStatus.CLIENT_BAD_REQUEST);
```

Comment:
This line is incorrect. It should return HttpStatus.SUCCESS_OK. 
The acceptor itself should return HttpStatus.CLIENT_BAD_REQUEST, but not because it calls doSessionOpened, but as a validation before actually calling doSessionOpen. Otherwise, we aren't actually checking the gateway but this newly created acceptHandler, and this is not what we want.

This comment applies to other tests in this commit as well.

@a-zuckut, @dpwspoon